Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -18,276 +17,286 @@
+
+        Returns:
+            dict: Response JSON.
+        """
+        return self._http_request(
+            method='POST',
+            url_suffix=f'/rest/2.0/export_profiles/{export_profile}/export',
+            raise_on_status=True,
+        )
+
+    def set_export_bookmark(self, export_profile: str) -> str:
+        """
+        Advances the internal bookmark / pointer for the export profile based on the last export events.
+        This is effectively an API-side implementation of `demisto.getLastRun` and `demisto.setLastRun`.
+
+        *Only call after successfully exporting events and sending them to XSIAM!*
+
+        Returns:
+            str: Response text.
+        """
+        return self._http_request(
+            method='POST',
+            url_suffix=f'/rest/2.0/export_profiles/{export_profile}/acknowledge',
+            raise_on_status=True,
+            resp_type='text',
         )
-        return response
 
 
-def test_module(client: Client, params: Dict[str, Any]) -> str:
+def test_module(client: Client, export_profiles: list[str]) -> str:
     """
-    Tests API connectivity and authentication'
-    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
-    successful.
-    Raises exceptions if something goes wrong.
+    Tests API connectivity and authentication and validates export profiles.
     Args:
         client (Client): Digital Guardian client to use.
-        params (Dict): Integration parameters.
+        export_profiles (list): List of export profile names.
     Returns:
-        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+        str: 'ok' connection to the service is successful.
+    Raises:
+        DemistoException | HTTPError: If request failed.
     """
-    try:
-        limit = arg_to_number(params.get('number_of_events')) or 1000
-        fetch_events(
-            client=client,
-            last_run={},
-            limit=limit,
-        )
+    invalid_export_profiles = set()
 
-    except Exception as e:
-        if 'Forbidden' in str(e):
-            raise DemistoException('Authorization Error: make sure API Key is correctly set')
-        else:
-            raise e
-    return "ok"
+    for export_profile in export_profiles:
+        try:
+            client.export_events(export_profile)
 
+        except Exception as e:
+            if 'Forbidden' in str(e):
+                raise DemistoException('Authorization Error: Make sure client credentials are correctly set') from e
 
-def get_raw_events(client: Client, time_of_last_event: str) -> list:
+            if 'Unable to find export_profile' in str(e):
+                invalid_export_profiles.add(export_profile)
+
+            else:
+                # Some other unknown error
+                raise
+
+    if invalid_export_profiles:
+        raise DemistoException(f'Invalid export profiles: {", ".join(invalid_export_profiles)}')
```

Comment:
nice. Since the `test instance` feature fails on each string that isn't `'ok'`, we may as well return this as a string rather than raise an exception, when we don't need the stack trace.

Reply:
Updated logic in commit de6eca13f444da175cf25459c9d3a91d521fdc81
We now return a non-`'ok'` string for known errors (authorization and invalid export profiles)